data/modules/TradeShips/Flow.lua

```diff
@@ -494,6 +494,25 @@ Flow.spawnInitialShips = function()
 	return ships_in_space
 end
 
+Flow.setPlayerAsTraderDocked = function()
+	local ship = Game.player
+	local dockedStation = ship:GetDockedWith()
+	if dockedStation then
+		Core.ships[ship] = { ts_error = "OK", status = 'docked', starport = dockedStation, ship_name = Game.player.shipId }
+		Trader.assignTask(ship, Game.time + utils.deviation(Core.params.port_params[Core.ships[ship].starport].time * 3600, 0.8), 'doUndock')
+	end
+end
+
+Flow.setPlayerAsTraderInbound = function()
+	local ship = Game.player
+	local nearestStation = ship:FindNearestTo("SPACESTATION")
+	if nearestStation then
+		Core.ships[ship] = { ts_error = "OK", status = 'inbound', starport = nearestStation, ship_name = Game.player.shipId }
+		Space.PutShipOnRoute(ship, Core.ships[ship].starport, Engine.rand:Number(0.0, 0.999))-- don't generate at the destination
```

Comment:
This function should currently be used immediately after exiting hyperspace - when the ship is in orbit around a star. I think it's worth either teleporting the player to the star, or writing a comment about the applicability of `setPlayerAsTraderInbound`.

Reply:
I will try to teleport player to star (or add comment if I fail), but what's so strange in current behaviour to do this step?

Reply:
If I understood the question correctly, for example, you described some strange behavior:
> And I don't know about Space.PutShipOnRoute. It frequently puts my ship in what looks like the center of the ground station's model, unless I haven't forgot something (or it's intended).

Reply:
Ah. I readed description for `Space.PutShipOnRoute` more closely and it says:
```
/*
 * Function: PutShipOnRoute
 *
 * The ship rearranged from its current position to a given body in space, for a
 * given part of the path, as if it were flying in a straight line, consuming
 * fuel and changing speed.
```
Basically, if `t_ratio` == 0.999, ship teleports to station's icon as you can see on `PutShipAtPointOfStation_Ground.mp4` video. This happened for me more frequently, because path between ship and station was short and as it says in the description:
**`The ship rearranged from its current position to a given body in space`**
But I already wrote `Space.PutShipIntoOrbit` on my computer, so I'm gonna include it in the next `force-push`.

Reply:
Added `PutShipIntoOrbit` in `LuaSpace.cpp`. Using this in the `Flow.setPlayerAsTraderInbound`.